### DISCUSSION TITLE
Provide a suggested checkpoint if rollback_to slot doesn't exist

### DISCUSSION THREAD

**Discussion by Quantumplation** [CardanoSolutions/kupo#90] [Archives]
#### Describe your idea, in simple words.

Suppose I want to track a new wallet, and I know the slot where I first sent it funds;

I don't want to provide *that* slot as the rollback point, because that will exclude the initial funds that were sent to the wallet.

So, I take the slot number (something like 79136895), and truncate it to a nice whole number (something like 79136000).  This fails.

My suggestion is to either:
 - provide an alternative payload that allows me to specify `ancestor_to` which uses the checkpoint to find the most recent ancestor
 - update the hint message that is returned to describe this, and include a suggested slot, such as

```json
{
  "hint":"The provided point (or slot) is unknown. Please provide an existing point from the past.  Either a block wasn't produced in that slot, or it is very recent. Note that if you provide a very recent point (e.g. 1-3 blocks old), it can happen that the point disappear should a rollback happen between the moment you looked it up and the moment you make this query.",
  "suggested_checkpoint": "79135995"
}
```

#### Why is it a good idea?
The first proposal would reduce the needed API calls by one, and save a round trip.
The second proposal would at least make the workflow, when doing things by hand, more obvious.

#### Are you willing to work on it yourself?
Yes, but as always, I doubt I'd have the time unless someone were to provide some hints at source files to look at. :sweat_smile:

**Comment by KtorZ** [comment]
Addressed by: https://github.com/CardanoSolutions/kupo/commit/14da8a30774a9d90d14be69ea1d2b91b7acf4cd8
